Classification/include/CGAL/Classification/classify.h

```diff
@@ -495,11 +495,11 @@ namespace internal {
       for (std::size_t y = 0; y < nb_y; ++ y)
       {
         bboxes.push_back
-          (CGAL::Bbox_3 (bbox.xmin() + Dx * (x / float(nb_x)),
-                         bbox.ymin() + Dy * (y / float(nb_y)),
+          (CGAL::Bbox_3 (bbox.xmin() + Dx * (x / double(nb_x)),
+                         bbox.ymin() + Dy * (y / double(nb_y)),
                          bbox.zmin(),
-                         bbox.xmin() + Dx * ((x+1) / float(nb_x)),
-                         bbox.ymin() + Dy * ((y+1) / float(nb_y)),
+                         bbox.xmin() + Dx * ((x+1) / double(nb_x)),
```

Comment:
I think you might not be able to achieve xmax/ymax due to rounding and you'll have an issue if the bound is tight. You should have a special case I guess.